testhelpers/binaries.go

```diff
@@ -0,0 +1,76 @@
+package testhelpers
+
+import (
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+)
+
+func ConditionallyBuildLifecycle(t *testing.T, dir string, goos string) {
+	AssertNil(t, os.MkdirAll(dir, 0755)) // Ensure directory exists
+
+	files, err := ioutil.ReadDir(dir)
+	AssertNil(t, err)
+
+	for _, file := range files {
+		if file.Name() == goos {
+			t.Log("WARNING: Not re-building lifecycle binaries, found existing binaries in", dir)
+			return
+		}
+	}
+
+	BuildLifecycle(t, dir, goos)
+}
+
+func BuildLifecycle(t *testing.T, dir string, goos string) {
+	cmd := exec.Command("make", "build-"+goos)
+	wd, err := os.Getwd()
+	AssertNil(t, err)
+	cmd.Dir = filepath.Join(wd, "..")
+	cmd.Env = append(
+		os.Environ(),
+		"GOOS="+goos,
+		"PWD="+cmd.Dir,
+		"BUILD_DIR="+dir,
+		"PLATFORM_API=0.9",
+		"LIFECYCLE_VERSION=some-version",
+		"SCM_COMMIT=asdf123",
+	)
+
+	t.Log("Building binaries: ", cmd.Args)
+	Run(t, cmd)
+}
+
+func CopyLifecycle(t *testing.T, src, dst string) {
```

Comment:
Do we really need it? Can't all tests read it from the directory we built it in?
And if we do need it, do we need to add it to `launcher_test.go`? Currently I see it only in `analyzer_test.go`.

Reply:
We need to copy it into the context for the `docker build`. Dockerfiles can't refer to files outside their context.

Reply:
Oh, interesting!
And what about `launcher_test.go`?

Reply:
I believe launcher_test.go is currently building the binaries directly into the container context. But I believe as discussed [here](https://github.com/buildpacks/lifecycle/pull/319#discussion_r445014704) we'll want to change the launcher tests to be consistent with what we're doing here.

Reply:
I think it's worth changing the launcher as part of this commit, since you introduce these functions, but I'm also ok with doing it in a separate commit.